tests/unit/pipeline/extractors/stores/test_splunk_extractor.py

```diff
@@ -0,0 +1,657 @@
+    mock_response = mocker.MagicMock()
+    mock_response.status_code = 200
+    mock_response.text = "Invalid response format"
+    mock_response.json.side_effect = json.JSONDecodeError("Not JSON", "", 0)
+
+    mock_client = mocker.MagicMock()
+    mock_client.get = AsyncMock(return_value=mock_response)
+
+    # Mock the logger to verify warning is logged
+    with patch.object(splunk_extractor, "logger") as mock_logger:
+        with pytest.raises(RuntimeError, match="Search job timed out"):
+            await splunk_extractor._wait_for_job_completion(
+                mock_client, "test123", max_wait_seconds=1
+            )
+
+        # Verify warning was logged for parse failure
+        mock_logger.warning.assert_called()
+        warning_call = mock_logger.warning.call_args
+        assert "Failed to parse job status" in warning_call[0][0]
+
+
+@pytest.mark.asyncio
+async def test_splunk_extractor_wait_for_job_completion_timeout(
+    splunk_extractor, mocker
+):
+    """Test job status checking timeout."""
+    mock_response = mocker.MagicMock()
+    mock_response.status_code = 200
+    mock_response.json.return_value = {
+        "entry": [{"content": {"dispatchState": "RUNNING"}}]
+    }
+
+    mock_client = mocker.MagicMock()
+    mock_client.get = AsyncMock(return_value=mock_response)
+
+    with pytest.raises(
+        RuntimeError, match="Search job timed out after 1 seconds: test123"
+    ):
+        await splunk_extractor._wait_for_job_completion(
+            mock_client, "test123", max_wait_seconds=1
+        )
+
+
+@pytest.mark.asyncio
+async def test_splunk_extractor_wait_for_job_completion_failure(
+    splunk_extractor, mocker
+):
+    """Test job status checking when job fails."""
+    mock_response = mocker.MagicMock()
+    mock_response.status_code = 200
+    mock_response.json.return_value = {
+        "entry": [{"content": {"dispatchState": "FAILED"}}]
+    }
+
+    mock_client = mocker.MagicMock()
+    mock_client.get = AsyncMock(return_value=mock_response)
+
+    with pytest.raises(RuntimeError, match="Search job failed"):
+        await splunk_extractor._wait_for_job_completion(mock_client, "test123")
+
+
+# Results retrieval tests
+@pytest.mark.asyncio
+async def test_splunk_extractor_get_job_results_single_chunk(
+    splunk_extractor, mock_job_results, mocker
+):
+    mock_response = mocker.MagicMock()
+    mock_response.status_code = 200
+    mock_response.json.return_value = mock_job_results
+
+    mock_client = mocker.MagicMock()
+    mock_client.get = AsyncMock(return_value=mock_response)
+
+    results = []
+    async for result in splunk_extractor._get_job_results(mock_client, "test123"):
+        results.append(result)
+
+    assert_that(results, has_length(3))
+    assert_that(
+        results[0],
+        has_entries(
+            {
+                "_time": "2023-01-01T10:00:00",
+                "host": "server1",
+                "message": "Login successful",
+            }
+        ),
+    )
```

Comment:
why not assert the whole result?

```suggestion
    results == [
      {
           "_time": "2023-01-01T10:00:00",
            "host": "server1",
            "message": "Login successful",
      }, 
      {...}, 
      {...}
    ]
```